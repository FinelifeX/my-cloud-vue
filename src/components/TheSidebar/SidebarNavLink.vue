<template>
  <router-link
    class="nav-link"
    active-class="nav-link--active"
    :exact="exact"
    :to="path"
  >
    <component :is="icon"></component>
    <span class="nav-link__text">{{ title }}</span>
  </router-link>
</template>

<script>
  export default {
    name: 'SidebarNavLink',
    props: {
      exact: {
        type: Boolean,
        default: true,
        required: false,
      },
      path: {
        type: String,
        required: true,
      },
      icon: {
        type: Object,
        default: null,
      },
      title: {
        type: String,
        required: true,
      },
    },
  };
</script>

<style scoped lang="scss">
  .nav-link {
    @apply flex flex-row items-center;
    @apply text-spun_pearl;
    @apply transition-colors duration-150;
    @apply pt-3 pb-3 pl-4 pr-4;
    @apply rounded-lg;

    &:hover {
      @apply text-indigo-600;
    }

    &--active {
      @apply text-indigo-600;
      @apply bg-selago;
      @apply font-semibold;
    }
  }

  .nav-link__text {
    @apply ml-6;
  }
</style>
