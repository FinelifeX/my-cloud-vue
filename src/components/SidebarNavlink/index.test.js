import { render } from '@testing-library/vue';
import SidebarNavLink from './index';

describe('SidebarNavLink component', () => {
  it('should render correctly', () => {
    const { html } = render(SidebarNavLink, {
      props: { title: 'Link', path: '/path' },
      routes: [],
    });

    expect(html()).toMatchSnapshot();
  });
});
