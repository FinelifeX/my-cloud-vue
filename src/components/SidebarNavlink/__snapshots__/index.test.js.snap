// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarNavLink component should render correctly 1`] = `
"<a href=\\"#/path\\" class=\\"nav-link\\">
  <!----> <span class=\\"nav-link__text\\">Link</span></a>"
`;
